source/common/ssl/ssl_socket.cc

```diff
@@ -36,11 +36,12 @@ Network::IoResult SslSocket::doRead(Buffer::Instance& read_buffer) {
   if (!handshake_complete_) {
     PostIoAction action = doHandshake();
     if (action == PostIoAction::Close || !handshake_complete_) {
-      return {action, 0};
+      return {action, 0, false};
```

Comment:
ditto here, w.r.t commenting the explicit false.

Reply:
I don't follow.  This one seems clear to me: we haven't read an EOF, so set "end_stream" to false.

Reply:
Not terribly SSL-savvy - are you asserting that doHandshake can never produce a SSL_ERROR_ZERO_RETURN in its internal error checking?   I didn't assume the handshake needed to be complete for the connection to be closed.

Reply:
Oh, I understand now.  Yes, I was assuming that the handshake can't return EOF.  However, I'm not sure if that's actually true or not.  I'll research and either comment or fix.

Reply:
nit: s/isnt't/isn't/

And today I've learned something new about SSL, thanks for doing the research :-)